traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
+    """ Event object to represent mutations to a list.
+
+    Attributes
+    ----------
+    trait_list : list
```

Comment:
Also, a design suggestion for this and the other collection change events: it might make sense to try to mimic the API of `TraitChangeEvent`.  At a minimum:

- using `object` instead of `trait_list`
- a dummy attribute `name` which is always `"items"`

This might allow code to consume both types of event without having to do a type check if it only cares about the object and the name.

That would imply an implicit "ChangeEvent" interface where you expect both "object" and "name" to exist.  It is tempting to make `new` be the new value of the list (ie. the same as `object`) and `old` to be a property that lazily reconstructs the old value of the list from the `added` and `removed`, but that isn't really right from a philosophical point of view.

Really the rest of the information is the "delta", ie. we have
- "who changed": the object that was changed, be it a HasTraits, TraitList, etc.
- "what changed": the name of the thing that changed (the thing that was being observed)
- "how it changed": the nature of the changes that took place (old/new, added/removed, etc.)

Thinking this way may help with the design of these classes.

Reply:
Hmmm we might need some use cases to drive this. For example, if a handler needs to consume `object` and `name`, the next thing it does may be to call `getattr(object, name)`, and doing that with `object` being `TraitList` and `name` being `"items"` is going to fail.

If the handler needs to do something specific about the change details, then it probably needs to know which type of event objects it gets anyway, it could go two routes:
(1) Apply `notify=True` only on the type of change that is compatible. Then it only deals with one type of event objects.
(2) Check the event type, and do different things according to the type.

Are there use cases already we can draw conclusions from?  Otherwise, leaving these `object` and `name` undefined for now might actually be more flexible, so we don't have to commit to a value before knowing how they are going to be used. e.g. `object` could be the `HasTraits` instance that contains the list somewhere (note that the list could be nested). It does not hurt if we end up having both `object` and `trait_list` pointing to the same thing.

Reply:
> "how it changed": the nature of the changes that took place (old/new, added/removed, etc.)

This is one of the reasons that I think it would make sense to bundle `index`, `removed` and `added` up into a single object, as I've suggested before. Those three pieces of information are always generated together, passed around together and used together.

Maybe open an issue to revisit this once the various related PRs are merged and we have a chance to look at the design as a whole?

Reply:
I am open to that discussion and it makes sense to do that at the very end instead of blocking these PRs. 
Do you mind opening the said issue please (since I am worried I may misunderstand the scope you are thinking about)?

Reply:
> Do you mind opening the said issue please

Opened #1084.